frame/election-provider-multi-phase/src/lib.rs

```diff
@@ -1642,6 +1726,11 @@ mod feasibility_check {
 				MultiPhase::feasibility_check(solution, COMPUTE),
 				FeasibilityError::SnapshotUnavailable
 			);
+
+			// kill also `SnapshotMetadata` and `DesiredTargets` for the storage state to be
+			// consistent for the try_state checks to pass.
```

Comment:
Another reason to hide these 3 storage items behind a shared wrapper stryct that will always manage their creation and destruction in synchrony :D 💡

Reply:
I agree, opened a new issue to handle this as a follow-up https://github.com/paritytech/substrate/issues/14032